posthog/tasks/email.py

```diff
@@ -502,3 +502,213 @@ def send_error_tracking_issue_assigned(assignment: ErrorTrackingIssueAssignment,
+    from posthog.models.hog_functions.hog_function import HogFunction
+
+    logger.info("Starting HogFunctions daily digest task")
+
+    # Calculate date range for yesterday's data
+    yesterday = timezone.now() - timedelta(days=1)
+    yesterday_start = yesterday.replace(hour=0, minute=0, second=0, microsecond=0)
+    yesterday_end = yesterday.replace(hour=23, minute=59, second=59, microsecond=999999)
+    date_str = yesterday.strftime("%Y-%m-%d")
+    formatted_date = yesterday.strftime("%B %d, %Y")
+
+    # Get all teams with active, non-deleted HogFunctions from PostgreSQL
+    teams_with_active_functions = (
+        HogFunction.objects.filter(enabled=True, deleted=False).values_list("team_id", flat=True).distinct()
+    )
+
+    team_ids = list(teams_with_active_functions)
+    logger.info(f"Found {len(team_ids)} teams with active HogFunctions")
+
+    if not team_ids:
+        logger.info("No teams with active HogFunctions found")
+        return
+
+    # Filter teams based on the feature flag setting
+    allowed_team_ids = settings.HOG_FUNCTIONS_DAILY_DIGEST_TEAM_IDS
+    if allowed_team_ids:
+        # Convert string team IDs to integers for comparison
+        allowed_team_ids_int = [int(team_id) for team_id in allowed_team_ids]
+        team_ids = [team_id for team_id in team_ids if team_id in allowed_team_ids_int]
+        logger.info(f"Filtered to {len(team_ids)} teams based on HOG_FUNCTIONS_DAILY_DIGEST_TEAM_IDS setting")
+
+        if not team_ids:
+            logger.info("No teams in allowed list have active HogFunctions")
+            return
+
+    # Get HogFunction data from PostgreSQL
+    hog_functions = HogFunction.objects.filter(team_id__in=team_ids, enabled=True, deleted=False).values(
+        "id", "team_id", "name", "type"
+    )
+
+    if not hog_functions:
+        logger.info("No active HogFunctions found for teams")
+        return
+
+    # Get metrics data from ClickHouse for these HogFunctions
+    hog_function_ids = [str(hf["id"]) for hf in hog_functions]
+
+    metrics_query = """
+    SELECT
+        team_id,
+        app_source_id as hog_function_id,
+        metric_name,
+        sum(count) as total_count
+    FROM app_metrics2
+    WHERE team_id IN %(team_ids)s
+    AND app_source = 'hog_function'
+    AND app_source_id IN %(hog_function_ids)s
+    AND timestamp >= %(start_time)s
+    AND timestamp <= %(end_time)s
+    AND metric_kind = 'other'
+    GROUP BY team_id, app_source_id, metric_name
+    ORDER BY team_id, app_source_id, metric_name
+    """
+
+    metrics_data = sync_execute(
+        metrics_query,
+        {
+            "team_ids": team_ids,
+            "hog_function_ids": hog_function_ids,
+            "start_time": yesterday_start,
+            "end_time": yesterday_end,
+        },
+    )
+
+    # Group data by team and calculate all aggregations
+    teams_data = {}
+
+    # First, initialize all HogFunctions
+    for hog_function in hog_functions:
```

Comment:
We have 100k hog functions.... This is going to be really slow. Also, we don't need to send an email for succeeding functions. I can't see why anybody would want that...

I think you should consider flipping this on its head:

1. Do a CH query to find all functions that have failures
2. Fan out and create a celery task per teamId containing the values of the failed functions
3. Have that task send the daily digest just detailing the ones that failed

If you disagree for whatever reason let me know and I will re-review but otherwise I don't think this will scale all that well.